source/widgets/ProjectStatus/js/configuration.js

```diff
@@ -29,13 +30,25 @@ function OctopusStatusWidgetConfiguration() {
                 return customSettings;
             };
 
-            var fetchDataSourceContent = function (queryUri, token, name, spaceId) {
-                let query;
+            var fetchDataSourceContent = function (queryUri, token, name, spaceId, partialNameFilter) {
+                let parameters = [];
                 if (spaceId) {
-                    query = '{"dataSourceDetails": {"dataSourceName":"' + name + '", "parameters": {"SpaceId":"' + spaceId + '"} } }';
+                    parameters.push('"SpaceId":"' + spaceId + '"');
+                }
+
+                // We check explicitly for `null` in this case. There are cases where we need to pass an empty string through to our endpointUrl.
+                if (partialNameFilter != null) {
+                    parameters.push('"PartialNameFilter":"' + partialNameFilter + '"');
+                }
+
+                let query;
+                if (parameters.length > 0) {
+                    let parametersCsv = "{" + parameters.join(", ") + "}";
+                    query = '{"dataSourceDetails": {"dataSourceName":"' + name + '", "parameters": ' + parametersCsv + " } }";
                 } else {
                     query = '{"dataSourceDetails": {"dataSourceName":"' + name + '"}}';
                 }
+                console.debug("query=", query);
```

Comment:
Was it intentional to leave this here? I can't see it hurting all that much; just confirming.